awscli/customizations/s3/s3handler.py

```diff
@@ -523,6 +525,80 @@ def _process_transfer(self, future):
             return CommandResult(1, 0)
 
 
+class S3TransferHandler(object):
+    def __init__(self, transfer_manager, cli_params, result_queue,
+                 result_recorder, result_processor):
+        """Backend for performing S3 transfers
+
+        :type transfer_manager: s3transfer.manager.TransferManager
+        :param transfer_manager: Transfer manager to use for transfers
+
+        :type cli_params: dict
+        :param cli_params: The parameters passed to the CLI command in the
+            form of a dictionary
+
+        :type result_queue: queue.Queue
+        :param result_queue: The queue to process results
+
+        :type result_recorder: ResultRecorder
+        :param result_recorder: Result recorder to get status of transfer
+
+        :type result_processor: ResultProcessor
+        :param result_processor: Result processor to process status updates
+            of transfer
+        """
+        self._transfer_manager = transfer_manager
+        self._result_queue = result_queue
```

Comment:
Possibly. So this is what I am planning. Let me know if that makes sense.

1) Change the interface of ``S3TransferHandler`` to:
```py
class S3TransferHandler(object):
    def __init__(self, transfer_manager, cli_params, result_queue):
```
2) Have the ``CommandResultRecorder`` live outside of ``S3TransferHandler`` because it eclipses more than the ``S3TransferHandler`` it covers the ``FileGenerators`` and any of the other abstractions. So I was going to have it do something like this:
```py
with command_result_recorder:
    fileinfos = filegenerator.call()
    s3_transfer_handler.call(fileinfos)
return command_result_recorder.get_command_result()
```